src/proto/mod.rs

```diff
@@ -326,3 +298,19 @@ impl LemmaProto {
         }
     }
 }
+
+/// The path of the lemma, as a binary string
+fn path_of_lemma(mut lemma: &Lemma) -> String {
```

Comment:
I would suggest having a custom return type instead of String. Such as a vector of bool wrapped in a newtype.
```
struct BinaryPath(Vec<bool>);
```
Adding a custom Display type will achieve your goal of succinct debug output.